dev/Breadcrumb/BreadcrumbItem.cpp

```diff
@@ -0,0 +1,263 @@
+
+        // Once an element has been clicked, close the flyout
+        CloseFlyout();
+
+        breadcrumbImpl->RaiseItemClickedEvent(senderAsContentControl.Content());
+    }
+}
+
+winrt::IInspectable BreadcrumbItem::CloneEllipsisItemSource(const winrt::Collections::IVector<winrt::IInspectable>& ellipsisItemsSource)
+{
+    // A copy of the hidden elements array in BreadcrumbLayout is created
+    // to avoid getting a Layout cycle exception
+    auto newItemsSource = winrt::make<Vector<winrt::IInspectable>>();
+
+    // The new list contains all the elements in reverse order
+    const int itemsSourceSize = ellipsisItemsSource.Size();
+    for (int i = itemsSourceSize - 1; i >= 0; --i)
+    {
+        const auto& item = ellipsisItemsSource.GetAt(i);
+        newItemsSource.Append(item);
+    }
+
+    return newItemsSource;
+}
+
+void BreadcrumbItem::OpenFlyout()
+{
+    if (auto flyout = m_ellipsisFlyout.get())
+    {
+        if (SharedHelpers::IsFlyoutShowOptionsAvailable())
+        {
+            winrt::FlyoutShowOptions options{};
+            flyout.ShowAt(*this, options);
+        }
+        else
+        {
+            flyout.ShowAt(*this);
+        }
+    }
+}
+
+void BreadcrumbItem::CloseFlyout()
+{
+    if (auto flyout = m_ellipsisFlyout.get())
+    {
+        flyout.Hide();
+    }
+}
+
+void BreadcrumbItem::OnEllipsisItemClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args)
+{
+    if (const auto& breadcrumb = m_parentBreadcrumb.get())
+    {
+        if (const auto& breadcrumbImpl = breadcrumb.try_as<Breadcrumb>())
+        {
+            const auto& hiddenElements = CloneEllipsisItemSource(breadcrumbImpl->HiddenElements());
+
+            if (const auto& flyoutRepeater = m_ellipsisItemsRepeater.get())
+            {
+                flyoutRepeater.ItemTemplate(m_ellipsisDataTemplate.get());
+                flyoutRepeater.ItemsSource(hiddenElements);
+            }
+
+            OpenFlyout();
+        }
+    }
+}
+
+void BreadcrumbItem::SetPropertiesForLastNode()
+{
+    m_isEllipsisNode = false;
+    m_isLastNode = true;
+
+    winrt::VisualStateManager::GoToState(*this, L"LastNode", false);
+}
+
+void BreadcrumbItem::ResetVisualProperties()
+{
+    m_isEllipsisNode = false;
+    m_isLastNode = false;
+
+    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        breadcrumbItemButton.Flyout(nullptr);
+    }
+    m_ellipsisFlyout.set(nullptr);
+    m_ellipsisItemsRepeater.set(nullptr);
+
+    winrt::VisualStateManager::GoToState(*this, L"Normal", false);
+}
+
+void BreadcrumbItem::InstantiateFlyout()
+{
+    // Only if the element has been created visually, instantiate the flyout
+    if (const auto& breadcrumbItemButton = m_breadcrumbItemButton.get())
+    {
+        // Create ItemsRepeater and set the DataTemplate 
+        const auto& ellipsisItemsRepeater = winrt::ItemsRepeater();
+        ellipsisItemsRepeater.Name(L"PART_EllipsisItemsRepeater");
```

Comment:
I don't think ellipsisItemsRepeater.Name(L"PART_EllipsisItemsRepeater") is needed.